### PR TITLE
Support skip data intervals

### PR DESCRIPTION
**Pull request by greycooker**
### PR types
New Feature
### PR changes
Support skip data intervals 
### Description
New training_arg skip_data_intervals, refer to the data intervals to skip, the training process will pass the data from start global step to end global step at each interval.

### CODE REVIEW

File: paddlenlp/trainer/trainer.py [hunk 1]
```diff
@@ -900,6 +902,7 @@ def _inner_training_loop(
             step_control = 0  # used in loop control, reset to 0 after every step
             self.control = self.callback_handler.on_epoch_begin(args, self.state, self.control)
 
+            step = -1
```
Comment (gongel) [inline paddlenlp/trainer/trainer.py:907]: 这里为什么要-1

File: paddlenlp/trainer/trainer_utils.py [hunk 2]
```diff
@@ -1065,3 +1065,20 @@ def set_hyrbid_parallel_seed(basic_seed, dataset_rank, tp_rank, pp_rank=0):
...
+
+    if skip_data_intervals is None:
+        return False
+    skip_flag = False
+    for interval in skip_data_intervals:
+        if len(interval) != 2 or interval[0] > interval[1] or interval[0] <= 0:
+            raise ValueError(f"Please check your skip interval {interval}")
+
+        start_global_step, end_step = interval[0], interval[1]
+        if start_global_step <= global_step + 1 <= end_step:
```
Comment (gongel) [inline paddlenlp/trainer/trainer_utils.py:1081]: global_step + 1的目的是什么？如果[5, 5]这种能跳过吗？

File: paddlenlp/trainer/trainer.py (hunk 1 above)
Comment (greycooker) [reply paddlenlp/trainer/trainer.py:907]: 这里是想修复之前的bug，如果不设置step=-1在这里
https://github.com/PaddlePaddle/PaddleNLP/blob/8f6b001e39f4d1600b2edc6110b657519de4bc07/paddlenlp/trainer/trainer.py#L1127C1-L1133C57可能会报错local variable 'step' referenced before assignment

File: paddlenlp/trainer/trainer_utils.py (hunk 2 above)
Comment (greycooker) [reply paddlenlp/trainer/trainer_utils.py:1081]: 比如设置start_global_step是1，那么是想跳过第一步的loss，那么在当前global_step=0的时候就应该跳数据

Comment (greycooker) [reply paddlenlp/trainer/trainer_utils.py:1081]: [5,5]这种能跳过

**Pull request closed without merging**